src/wsproto/handshake.py

```diff
@@ -236,14 +236,6 @@ def _process_connection_request(  # noqa: MC0001
             raise RemoteProtocolError(
                 "Missing header, 'Upgrade: WebSocket'", event_hint=RejectConnection()
             )
-        if version is None:
-            raise RemoteProtocolError(
-                "Missing header, 'Sec-WebSocket-Version'", event_hint=RejectConnection()
-            )
-        if host is None:
```

Comment:
`host` cannot be None here, because the h11 Request would have already raised an exception.

Reply:
Haven't looked at the code but just as a heads up, h11 allows incoming connections to not have host headers, as long as they're declared as being http/1.0. Not sure if that's relevant here or not.

Reply:
Thanks! I did add a test to check this behaviour: https://github.com/python-hyper/wsproto/blob/400e1db3a075fdc60a028d26f590b43f58f4d2aa/test/test_handshake.py#L29-L30
Not sure if we explicitly set HTTP/1.1 somewhere, but at least we would catch it if something would change.